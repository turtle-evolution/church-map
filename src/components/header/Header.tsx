import { useHistory } from 'react-router-dom';

import {
  EuiHeader,
  EuiHeaderLink,
  EuiHeaderLinks,
  EuiHeaderLogo,
  EuiHeaderSectionItem,
  useEuiI18n
} from '@elastic/eui';
import useIcon from 'hook/useIcon';

const Header = (): JSX.Element => {
  const iconVisMapRegion = useIcon('visMapRegion');
  const history = useHistory();

  const goHome = () => {
    history.push('/');
  };

  return (
    <EuiHeader>
      <EuiHeaderSectionItem border="right">
        <EuiHeaderLogo
          iconType={iconVisMapRegion}
          data-testid="logo"
          onClick={goHome}
        >
          {useEuiI18n('euiContext.location', 'Location')}
        </EuiHeaderLogo>
      </EuiHeaderSectionItem>
      <EuiHeaderSectionItem>
        <EuiHeaderLinks aria-label="App navigation links example">
          <EuiHeaderLink isActive>Docs</EuiHeaderLink>
          <EuiHeaderLink>Code</EuiHeaderLink>
          <EuiHeaderLink>Help</EuiHeaderLink>
        </EuiHeaderLinks>
      </EuiHeaderSectionItem>
    </EuiHeader>
  );
};

export default Header;
