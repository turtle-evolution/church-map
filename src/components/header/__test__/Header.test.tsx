import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import Header from '../Header';

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('Header Component', () => {
  test('should take a snapshot', () => {
    const { asFragment } = render(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  test('on click logo', () => {
    render(<Header />);
    const logo = screen.getByTestId('logo');
    fireEvent.click(logo);

    expect(mockHistoryPush).toBeCalled();
  });
});
