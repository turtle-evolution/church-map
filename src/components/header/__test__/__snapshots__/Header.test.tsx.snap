// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="euiHeader euiHeader--default euiHeader--static"
  >
    <div
      class="euiHeaderSectionItem euiHeaderSectionItem--borderRight"
    >
      <a
        class="euiHeaderLogo"
        data-testid="logo"
        rel="noreferrer"
      >
        <svg
          aria-hidden="true"
          aria-label="Elastic"
          class="euiIcon euiIcon--large euiIcon-isLoading euiHeaderLogo__icon"
          focusable="false"
          height="16"
          role="img"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        />
        <span
          class="euiHeaderLogo__text"
        >
          Location
        </span>
      </a>
    </div>
    <div
      class="euiHeaderSectionItem euiHeaderSectionItem--borderLeft"
    >
      <nav
        aria-label="App navigation links example"
        class="euiHeaderLinks"
      >
        <div
          class="euiHeaderLinks__list euiHeaderLinks__list--gutterS"
        >
          <button
            class="euiButtonEmpty euiButtonEmpty--primary euiHeaderLink euiHeaderLink-isActive"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Docs
              </span>
            </span>
          </button>
          <button
            class="euiButtonEmpty euiButtonEmpty--text euiHeaderLink"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Code
              </span>
            </span>
          </button>
          <button
            class="euiButtonEmpty euiButtonEmpty--text euiHeaderLink"
            type="button"
          >
            <span
              class="euiButtonContent euiButtonEmpty__content"
            >
              <span
                class="euiButtonEmpty__text"
              >
                Help
              </span>
            </span>
          </button>
        </div>
      </nav>
    </div>
  </div>
</DocumentFragment>
`;
